### ISSUE TITLE
Get routing still not working in version 4.6.6

### ISSUE THREAD

**Issue by CarlosBernalA** [jonataslaw/getx#2918]
When navigating to a route with Get.toName takes you to the route correctly, but when leaving the route with Get.Back or any other means the state of the route remains and when entering the route again it remains with the previous state, The Controller continues with the same state as the previous one, it does not die at any time, it is still in memory.

A few months ago I was forced to make some modifications to the get code, temporarily while another version comes out, but the modification was simple but I am not sure it is the correct change.
The change I made was to the router_observer.dart file in the didPop event.
Below I attach a screenshot, maybe it will help:
![Codigo](https://github.com/jonataslaw/getx/assets/57161416/a4f93715-4805-4698-abb0-ed3aa62ce3b9)

Getx Version:
get: ^4.6.6

**Comment by CarlosBernalA** [comment]
Note that this modification partially solved the problem.

**Comment by mohsinnaqvi0606** [comment]
This issue only exist in **unNamed** routing. I tried **Named** routing and it's working properly

Define route like this in GetMaterialApp

```
initialRoute: '/',
      getPages: [
        GetPage(name: '/', page: () => Splash()),
      ],
```
Where you want to navigation write this code
`Get.offNamed('/');`
OR
`Get.toNamed('/');`

**Comment by mohsinnaqvi606** [comment]
@jonataslaw , @CarlosBernalA
This issue only exist in **unNamed** routing. I tried **Named** routing and it's working properly

Define route like this in **GetMaterialApp**

```
initialRoute: '/',
      getPages: [
        GetPage(name: '/', page: () => Splash()),
      ],
```
Where you want to navigation write this code
`Get.offNamed('/');`
OR
`Get.toNamed('/');`

**Comment by CarlosBernalA** [comment]
@mohsinnaqvi606 Even so, the problem persists, this happened after updating Flutter to version 3, before that there was no problem, I attach images of the initialization of the routung of my app.
![image](https://github.com/jonataslaw/getx/assets/57161416/ed23f122-c9d7-4735-b41e-7288593bbd01)
![image](https://github.com/jonataslaw/getx/assets/57161416/7e8e637c-25a8-45d3-ae31-2a022ebd151b)


When calling Get.back() or another back method it only shows this message in the console, before it eliminated memory
![image](https://github.com/jonataslaw/getx/assets/57161416/7d36331e-6d9a-449d-ba18-cf822bca2572)

**Comment by mohsinnaqvi0606** [comment]
> @mohsinnaqvi606 Even so, the problem persists, this happened after updating Flutter to version 3, before that there was no problem, I attach images of the initialization of the routung of my app. ![image](https://user-images.githubusercontent.com/57161416/269013575-ed23f122-c9d7-4735-b41e-7288593bbd01.png) ![image](https://user-images.githubusercontent.com/57161416/269013717-7e8e637c-25a8-45d3-ae31-2a022ebd151b.png)
> 
> When calling Get.back() or another back method it only shows this message in the console, before it eliminated memory ![image](https://user-images.githubusercontent.com/57161416/269014564-7d36331e-6d9a-449d-ba18-cf822bca2572.png)

@CarlosBernalA Yes, you are right. It's working only in few cases. It's hard to understand, what is the cause of issue.
My suggested solution worked in one project. but it's still not working in other project

**Comment by mario987111** [comment]
You're right. I had the same problem

**Comment by KamalBalah** [comment]
same issue
